Sources/Datadog/OpenTracing/OTSpan.swift

```diff
@@ -74,3 +74,56 @@ public extension OTSpan {
         self.finish(at: Date())
     }
 }
+
+/// Error conveniences
+public extension OTSpan {
+    /// Set or replace the error for the given span.
+    /// This is a convenience to set the proper tags with error details. Consider the `setError(message:stacktrace:file:line:)` variant for a better control over the error details.
+    ///
+    /// - parameter error: An object conforming to the `Error` protocol.
+    /// - parameter file: A string identifying the file where the `Error` was caught. The default is `#fileID` which means `ModuleName/Filename.extension`, consider an helpful yet concise identifier when overriding the default.
+    /// - parameter line: The line number in the file where the `Error` was caught.
+    func setError(
+        _ error: Error,
+        file: StaticString = #fileID,
+        line: UInt = #line
+    ) {
+        let dderror = DDError(error: error)
+        setError(
+            kind: dderror.title,
+            message: dderror.message,
+            stack: dderror.details,
+            file: file,
+            line: line
+        )
+    }
+
+    /// Set or replace the error for the given span.
+    /// This is a convenience to set the proper tags with error details.
+    ///
+    /// - parameter kind: The type of error to be logged.
+    /// - parameter message: An error message to be logged.
+    /// - parameter stack: A string detailing the state of the stack when the error was caught. Note that it can also be any details that could help further triaging and investigation of the error downstream, it doesn't have to be an actual stack trace.
+    /// - parameter file: A string identifying the file where the error was caught. The default is `#fileID` which means `ModuleName/Filename.extension`, consider an helpful yet concise identifier when overriding the default.
+    /// - parameter line: The line number in the file where the error was caught.
+    func setError(
+        kind: String,
+        message: String,
+        stack: String = "",
+        file: StaticString = #fileID,
+        line: UInt = #line
+    ) {
+        var stackWithFile = "\(file):\(line)"
+        if stack.count > 0 {
+            stackWithFile += "\n" + stack
+        }
+        log(
```

Comment:
As this uses Logging feature under the hood, maybe we can add a note on its availability in those 2 API comments? Otherwise, if someone configures the SDK with explicitly disabling logging: `.enableLogging(false)`, this error won't be send and the warning printed to the console will be:
```
"The log for span <span> will not be send, because the Logging feature is disabled."
```
This message makes it unclear: what log if `span.setError()` API was called.

Reply:
Maybe it's confusing but I think this `log()` API is part of OpenTracing (especially `OTSpan`) and only requires to enable tracing, which if the user is being consistent should already be enabled prior to using `OTSpan`/`DDSpan`.
Is my assumption correct? Happy to add the comment otherwise.

Reply:
i'm _slightly_ for adding a note at the end of comments
and the same for `setError(_ error: Error...)` above

Reply:
> Maybe it's confusing but I think this log() API is part of OpenTracing (especially OTSpan) and only requires to enable tracing, which if the user is being consistent should already be enabled prior to using OTSpan/DDSpan.
Is my assumption correct?

Yes `span.log()` is part of Open Tracing API. However, our implementation of this API uses the `Logging` feature under the hood. When the `span.log()` is called, we send a real `Log` to Datadog Logging intake, using integration with `LoggingFeature` (this `Log` contains `spanID` and `traceID`, so we can correlate it on backend).

This means that initializing the SDK with `.enableLogging(false)` and using `span.log()` will print the warning to the user. The warning is defined here:

https://github.com/DataDog/dd-sdk-ios/blob/71b46468396adbd101adc9a4309a5707ad43e4ce/Sources/Datadog/Tracer.swift#L209-L216

and is part of the control flow executed for `span.log()`.

Reply:
Got it! I'll add a comment about that.